AsyncDisplayKit/ASViewController.m

```diff
@@ -0,0 +1,47 @@
+//
+//  ASViewController.m
+//  AsyncDisplayKit
+//
+//  Created by Huy Nguyen on 16/09/15.
+//  Copyright (c) 2015 Facebook. All rights reserved.
+//
+
+#import "ASViewController.h"
+#import "ASAssert.h"
+#import "ASDimension.h"
+
+@implementation ASViewController
+
+- (instancetype)initWithNode:(ASDisplayNode *)node
+{
+  if (!(self = [super init])) {
+    return nil;
+  }
+  
+  ASDisplayNodeAssertNotNil(node, @"Node must not be nil");
+  ASDisplayNodeAssertTrue(!node.layerBacked);
+  _node = node;
+  
+  return self;
+}
+
+- (void)loadView
+{
+  ASDisplayNodeAssertTrue(!_node.layerBacked);
+  self.view = _node.view;
+}
+
+- (void)viewWillLayoutSubviews
+{
+  ASSizeRange constrainedSize = ASSizeRangeMake(self.view.bounds.size, self.view.bounds.size);
+  [_node measureWithSizeRange:constrainedSize];
```

Comment:
Cool, I think this effectively does synchronous measurement here and means that just barely in time for -layout, there will be cached sizes.

What would happen if measureWithSizeRange were called on a background thread (on the node) and were still running at this point?  Interestingly there haven't been other cases in usage (either framework or app) where I have needed to consider that.

Lastly, small nit, but always create a local variable for a structure that needs to be accessed twice.  CGSize boundsSize = self.view.bounds.size;